gitless/tests/test_e2e.py

```diff
@@ -683,6 +684,86 @@ def test_uncommitted_tracked_changes_that_conflict_append(self):
     self.assertTrue('contents 2' in contents)
 
 
+class TestEmptyDir(TestEndToEnd):
+
+  def test_empty_dir_status(self):
+    untracked_empty_dir = self._mk_empty_dir('untracked_empty_dir')
+
+    out = utils.stdout(gl.status())
+
+    self.assertIn(untracked_empty_dir, out, 'Empty dir didn\'t appear in status')
+
+  def test_ignored_empty_dir_status(self):
+    ignored_empty_dir = self._mk_empty_dir('ignored_empty_dir')
+    utils.write_file(os.path.join(self.path, '.gitignore'), ignored_empty_dir)
+
+    out = utils.stdout(gl.status())
+
+    self.assertFalse(ignored_empty_dir in out,
+        'Ignored empty dir was listed in status')
+
+  def test_track_empty_dir(self):
+    dir_to_track = self._mk_empty_dir('wanted_empty_dir')
+    expected_out = 'Empty directory {0} is now a tracked directory'.format(
+      self._dir_path(dir_to_track))
+
+    out = utils.stdout(gl.track(dir_to_track))
+
+    self.assertIn(expected_out, out, 'Empty dir wasn\'t tracked')
+
+  def test_track_parent_empty_dir(self):
+    parent_empty_dir = self._mk_empty_dir('parent_empty_dir')
+    child_dir = self._mk_empty_dir(os.path.join(parent_empty_dir, 'child_dir'))
+    unexpected_out = 'Empty directory {0} is now a tracked directory'.format(
+        os.path.join(child_dir, ''))
+
+    out = utils.stdout(gl.track(parent_empty_dir))
+
+    self.assertFalse(unexpected_out in out, 'Tracked empty dir child')
+
+  def test_tracked_empty_dir_status(self):
+    tracked_empty_dir = self._mk_empty_dir('tracked_empty_dir')
+    gl.track(tracked_empty_dir)
+    expected_out = '{0} (new directory)'.format(
+        self._dir_path(tracked_empty_dir))
+
+    out = utils.stdout(gl.status())
+
+    self.assertIn(expected_out, out, 'Didn\'t report newly tracked dir')
+
+  def test_commit_empty_dir(self):
+    empty_dir = self._mk_empty_dir('wanted_empty_dir')
+    gl.track(empty_dir)
+    gl.commit(_out='std.out', _bg=True)
+    while(sum(1 for line in open('std.out')) < 7):
```

Comment:
Couldn't find a better way for checking when commit dialogue is completely written to text editor. 😕